### PR TITLE
add user agent transformer

### PR DESCRIPTION
**Pull request by wonderflow**
## Changes
   
  - [ ] feature1
  - [ ] feature2 
  - [ ] fixbug1
  - [ ] fixbug2
 
## Reviewers

  - [ ] @[someone] please review
  - [ ] @[someotherone] please review

## Wiki Changes
 
  - options1...
  - options2...
    
## Checklist
   
   - [ ] Rebased/mergable
   - [ ] Tests pass
   - [ ] Wiki updated

### CODE REVIEW

**Comment by andrewei1316** [comment]
这个里面还用到了 yaml 库？

**Comment by wonderflow** [comment]
用到了

File: transforms/ua/ua.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+			err = fmt.Errorf("transform key %v not exist in data", it.Key)
+			continue
+		}
+		strval, ok := val.(string)
+		if !ok {
+			errnums++
+			err = fmt.Errorf("transform key %v data type is not string", it.Key)
+			continue
+		}
+		client := it.uap.Parse(strval)
```
Comment (andrewei1316) [inline transforms/ua/ua.go]: `strval == ""` 的时候是什么行为？

Comment (wonderflow) [reply transforms/ua/ua.go]: 解析出来的都是 Other，Other，Other

**Comment by andrewei1316** [comment]
LGTM

**Pull request merged**